### PR TITLE
create Pull Request template

### LINKED ISSUE #20: create Pull Request template
Whoever is reading this, plz note that I need a detailed PR template asking for:
- why this change is required
- is this tested, how?
- will this affect any functionality?
- is this backward compatible?

something like this and in detail... this is just an example... and plz ask to contribute first... I'll assign it to you, otherwise it might get assigned to someone else or maybe I'll do it myself, which will result in wasting your own time :slightly_smiling_face: So plz do LMK before doing it

Thanks

### PR DESCRIPTION
**Pull request by Mastersam07**
Closes #20

- added pull request template

### CODE REVIEW

**Comment by iamdipanshusingh** [comment]
@Mastersam07 plz put the template inside the `.github` folder and also comment out a few things which will be only necessary to read when somebody's actually creating a PR... rest is fine

**Comment by Mastersam07** [comment]
@iamdipanshusingh I have made some new changes. I am wondering how adding this template reduces the codecov %

**Comment by iamdipanshusingh** [comment]
@Mastersam07 plz ignore it... I think it's cuz you were on an older version which had 100% coverage... which is now reduced to somewhere around... so it doesn't matter

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -22,18 +22,20 @@ Please describe the tests that you ran to verify your changes. Provide instructi
...
+<!---
 **Test Configuration**
 * Hardware:
 * Flutter SDK Version:
 * Flutter Doctor Results:
-
+-->
 ## Is This Backward Compatible?
 
 Are you changes backwards compatible?
```
Comment (iamdipanshusingh) [inline .github/pull_request_template.md]: @Mastersam07 plz fix the typo here... it should be `Are your changes backward compatible?`

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Pull Request Template
+
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. State why this change is required and list any dependencies that are required for this change.
```
Comment (iamdipanshusingh) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: @Mastersam07 plz comment it out... not needed unnecessarily

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Pull Request Template
```
Comment (iamdipanshusingh) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: @Mastersam07 can this be changed to `Hey, I just made a Pull Request` or maybe something else but not this :sweat_smile:

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Test A
+- [ ] Test B
```
Comment (iamdipanshusingh) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: this body can be commented out as well

File: .github/pull_request_template.md (hunk 1 above)
Comment (Mastersam07) [reply .github/pull_request_template.md]: Oh, thanks

**Review by iamdipanshusingh — CHANGES_REQUESTED** [review]
@Mastersam07 have requested some minor changes, sorry for asking such things like this.. :sweat_smile:

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 3 above)
Comment (Mastersam07) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Okay 😄

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 2 above)
Comment (Mastersam07) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Okay 👍

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 4 above)
Comment (Mastersam07) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Okay 👍

**Comment by Mastersam07** [comment]
Updated @iamdipanshusingh

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,8 +1,10 @@
-# Pull Request Template
+# Hey, i just made a pull request ...
```
Comment (iamdipanshusingh) [inline .github/pull_request_template.md]: @Mastersam07 typo here.... `I` and instead of `...` an `!` would be enough

**Review by iamdipanshusingh — APPROVED** [review]
LGTM :+1:

**Comment by Mastersam07** [comment]
@iamdipanshusingh changes made

**Comment by iamdipanshusingh** [comment]
@Mastersam07 plz feel free to contribute even more... docs, features, test cases, or anything else

**Pull request merged**